generator/src/main/java/org/apache/kafka/message/checker/CheckerUtils.java

```diff
@@ -91,4 +102,54 @@ static MessageSpec readMessageSpecFromFile(String schemaPath) {
             throw new RuntimeException("Unable to parse file as MessageSpec: " + schemaPath, e);
         }
     }
+
+    /**
+     * Return a MessageSpec file give file contents.
+     *
+     * @param contents      The path to read the file from.
+     * @return              The MessageSpec.
+     */
+    static MessageSpec readMessageSpecFromString(String contents) {
```

Comment:
we don't need to create a file to translate a string to a MessageSpec, we can do this directly.